docs/manifest.md

```diff
@@ -128,17 +128,23 @@ To enable this feature, you need to declare the modified files or components usi
 - `--modified-files`
 - `--modified-components`
 
-To declare the dependencies for an app, you could add two more keywords in the corresponding manifest file:
+By default, one app will be built when:
+
+- The app itself is modified (`.md` files are excluded)
+- Any of the `build_components` are modified.
+
+    `build_components` are defined in the `project_description.json`, which is generated by running `idf.py reconfigure` or `idf.py build`
+
+To declare the dependencies for an app explicitly, `idf-build-apps` supports two more manifest file entries:
 
 - `depends_components`
 - `depends_filepatterns`
 
-One app will be built when:
+When any of these two entries are specified, the default behavior will be different. Instead, the app will be built when:
 
 - The app itself is modified (`.md` files are excluded)
-- Any of the specified `depends_components` in the corresponding manifest file are modified
-- Files that matches any of the specified `depends_filepatterns` in the corresponding manifest file are modified
-- Any of the `build_components` are modified. `build_components` are defined in the `project_description.json`, which is generated by running `idf.py reconfigure` or `idf.py build`
+- Any of the specified `depends_components` is modified
+- Any file that matches any of the specified `depends_filepatterns` is modified
 
```

Comment:
Should we write it explicitly that when using these options `build_components` no longer affect the build? Just to make it 100% clear.

Reply:
I think we can make this clear by restructuring the description a bit:

---
## Building apps only on related changes

In large projects or monorepos it is often desirable to only run builds and tests which are somehow related to the changes in a pull request.

`idf-build-apps`  supports this by checking whether a particular app has been modified, or depends on modified components or modified files.

### Basic Usage

To enable this feature, you need to pass the list of modified files or modified components to idf-build-apps using the following CLI options:

- `--modified-files`
- `--modified-components`

For example, if the project uses Git, you can obtain the list of files modified in a branch or a pull request by calling
```bash
git diff --name-only ${pr_branch_head} $(git merge-base ${pr_branch_head} main)
```
where `pr_branch_head` is the branch of the pull request, and `main` is the default branch.

idf_build_apps uses the following rules to determine whether to build an app or not:

1. Any files in the app itself are modified (`.md` files are excluded)
2. If `depends_components` or `depends_filepatterns` are specified in the manifest file, idf-build-apps matches `modified-components` and `modified-files` against these two options. If any of the modified components are in the `depends_components` list or any of the modified files are matched by `depends_filepatterns`, the app is built.
3. If `depends_components` or `depends_filepatterns` are not specified in the manifest file, idf-build-apps determines the list of components the app depends on using `BUILD_COMPONENTS` property in IDF build system. For the given app, this property contains the list of all the components included into the build. idf_build_apps runs `idf.py reconfigure` to determine the value of this property for the app. If any of the modified components are present in BUILD_COMPONENTS list, the app is built.